src/js/containers/search/filters/awardID/AwardIDListContainer.jsx

```diff
@@ -0,0 +1,209 @@
+    parent_award__piid: React.PropTypes.array
+};
+
+export class AwardIDListContainer extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            awardIDSearchString: '',
+            autocompleteAwardIDs: [],
+            noResults: false
+        };
+
+        this.handleTextInput = this.handleTextInput.bind(this);
+        this.clearAutocompleteSuggestions = this.clearAutocompleteSuggestions.bind(this);
+        this.timeout = null;
+        this.fields = ['piid', 'fain', 'uri', 'parent_award__piid'];
+    }
+
+    componentDidMount() {
+        this.parseAutocompleteAwardIDs(this.props);
+    }
+
+    componentWillReceiveProps(nextProps) {
+        if (!_.isEqual(nextProps.piid, this.props.piid)
+            || !_.isEqual(nextProps.fain, this.props.fain)
+            || !_.isEqual(nextProps.uri, this.props.uri)
+            || !_.isEqual(nextProps.parent_award__piid, this.props.parent_award__piid)) {
+            this.parseAutocompleteAwardIDs(nextProps);
+        }
+    }
+
+    parseAutocompleteAwardIDs(awardIDProps) {
+        const values = [];
+        const fieldMap = {
+            piid: 'PIID',
+            fain: 'FAIN',
+            uri: 'URI',
+            parent_award__piid: 'PIID'
+        };
+
+        Object.keys(fieldMap).forEach((key) => {
+            const awardIDs = awardIDProps[key];
+            if (awardIDs && awardIDs.length > 0) {
+                awardIDs.forEach((item) => {
+                    if (item) {
+                        values.push({
+                            title: AwardIDFormatter.formatAwardID(item, fieldMap[key]),
+                            subtitle: fieldMap[key],
+                            data: Object.assign({}, item, {
+                                awardIDType: fieldMap[key]
+                            })
+                        });
+                    }
+                });
+            }
+        });
+
+        this.setState({
+            autocompleteAwardIDs: values
+        });
+    }
+
+    queryAutocompleteAwardIDs(input) {
+        this.setState({
+            noResults: false
+        });
+
+        // Only search if input is 2 or more characters
+        if (input.length >= 2) {
+            this.setState({
+                awardIDSearchString: input
+            });
+
+            if (this.awardIDSearchRequest) {
+                // A request is currently in-flight, cancel it
+                this.awardIDSearchRequest.cancel();
+            }
+
+            const awardIDSearchParams = {
+                fields: this.fields,
+                value: this.state.awardIDSearchString,
+                matched_objects: true
+            };
+
+            this.awardIDSearchRequest = SearchHelper.fetchAwardIDs(awardIDSearchParams);
+
+            this.awardIDSearchRequest.promise
+                .then((res) => {
+                    this.setState({
+                        noResults: !res.data.matched_objects.length
```

Comment:
I don't think this is correctly setting the flag. Maybe an explicit `res.data.matched_objects.length === 0`?